### DISCUSSION TITLE
BRouter (offline) turn instruction improvements

### DISCUSSION THREAD

**Discussion by REHimalayan** [devemux86/cruiser#622] [Q&A] [answered]
Hallo zusammen,
ich habe heute Cruiser bei einer Rundtour getestet und finde die Audioausgabe gewöhnungsbedürftig. An so ziemlich jeder Straßenkreuzung soll ich mich "leicht links" halten, obwohl ich doch eigentlich geradeaus fahren will. Warum kommt da überhaupt eine Meldung? 
Wenn ich wirklich abbiegen soll, ist das auch nur "leicht links" oder "leicht rechts". Zumindest in 8 von 10 Fällen. 
Habe ich aus Versehen etwas verstellt oder muss das so sein? Unten das Bild so einer "leicht rechts" Kurve....

Ansonsten bin ich begeistert von Cruiser aber da mein Telefon nicht wirklich in der Sichtachse montiert ist, bin ich ziemlich auf die Audio navigation angewiesen.

<img src="https://github.com/user-attachments/assets/5e944f74-08d5-4a5c-bacc-8c9f9f31e4d1" width="50%"/>

**Comment by r11d11** [comment]
Hallo REHimalayan,
du hättest besser dabei geschrieben, wo genaiu der Abzweig ist.
Ich denke es ist dieser:

https://plus.codes/9F2C6FJ5+CG

Bin überrascht aver da kommt bei mir tatsächlich auch  "leicht rechs" obwohl das eher nach scharf rechts aussieht.
Die scharf rechts Anweisungen klappen normalerweise auch. 
Liegt wahrscheinlich an dieser Stelle an den OSM Karten.
Routest du mit Brouter?
Dann kommen die Anweisung von Brouter und Cruiser kann nur das ansagen, was Brouter berechnet.

Und du hast recht. Wenn man geradeausfahren soll bekommt mat tasächlich oft die Meldung "halten sie sie sich links oder rechts" 
Ich gucke dann auf die Karte und fahre geradeaus.
Man gewöhnt sich dran. 

Die  Ansagen von Brouter sind teilweise tatsächlich nicht optimal aber ich bin meistens zufrieden.
Nur nach den Ansagen zu fahren würde aber auch bei mir nicht gut funktionieren.
Ich brauche immer den Bildschirm und die Ansage.
Dann klappt es auch zu 99%

Du darfst aber auf jedenfall keine so präzisen Anweisungen wie bei Google Maps erwarten.
Ich habe ganz schön dumm geguckt als ich zum ersten Mal auf der Autobahn war und die Ausfahrt nehmen sollte und dann
kam die Meldung "halten Sie sich rechts".  Und das passiert bei jeder Ausfahrt.
Das finde ich schon enttäuschend. Aber habe mich dran gewöhnt.

Aber auch wenn die Ansagen von Cruiser/Brouter nicht optimal sind hat Criuser soviele andere Vorteile, dass ich nur noch mit Cruiser navigiere.

**Comment by devemux86** [comment]
@REHimalayan 

Thanks for the report!
- What routing service do you use, BRouter (offline)?

- Can you post here the route so we can check it?
Export it as GPX (navigation) or JSON. Or use [BRouter website](https://brouter.de/brouter-web/).

---

Vielen Dank für den Bericht!

- Welchen Routing-Dienst verwenden Sie, BRouter (offline)?

- Können Sie die Route hier posten, damit wir sie überprüfen können?
Exportieren Sie sie als GPX (Navigation) oder JSON. Oder verwenden Sie die [BRouter-Website](https://brouter.de/brouter-web/).

**Comment by REHimalayan** [comment]
Thanks for your replys!

I think I should try to switch to english for my postings.

I use BRouter (offline). If I switch to OpenRouteService the voice command is as expected, "turn right". Also the "stay left/stay right" on crossings are gone. The voice commands from BRouter (online) are also correct. So maybe there is an "issue" with BRouter (offline) or I have to get used to...


[slight_right.json](https://github.com/user-attachments/files/16643904/slight_right.json)
[keep_right.json](https://github.com/user-attachments/files/16643905/keep_right.json)

**Reply by devemux86** [reply]
@REHimalayan Thanks for the route examples!

The turn instructions are provided by the routing services, here by BRouter (offline).

You can also check them with the menu "Routing | Turn instructions"
and on the map if you enable their markers in "Settings | Routing | Appearance".

I also created an issue in BRouter repository to discuss it:
- https://github.com/abrensch/brouter/issues/723

<img src="https://github.com/user-attachments/assets/43dca7cd-2358-41a1-aede-59564f303ae9" width="50%"/><img src="https://github.com/user-attachments/assets/5c428be0-7c07-4b4c-ad97-dc10ee6c750d" width="50%"/>

**Comment by devemux86** [comment]
BRouter improvements in turn instructions were merged and will be available in the next Cruiser.

BRouter discussion with the reports:
- https://github.com/abrensch/brouter/issues/723

**Comment by devemux86** [comment]
BRouter improvements in turn instructions are available in **[Cruiser 5.1.0-b6](https://github.com/devemux86/cruiser/discussions/671)**.

---

Remember that the turn instructions are generated by the router.

It is best to report them directly to the BRouter repository with [BRouter website](https://brouter.de/brouter-web/) route links:
- https://github.com/abrensch/brouter/issues/723

**Comment by devemux86** [comment]
New BRouter instructions for "**Exit left**" / "**Exit right**" are available in the latest Cruiser.

Copying here the BRouter definition for the motorway exit:
_"junction on motorway via motorway_link and next way less then motorway_link is a motorway exit"_

BRouter discussion with the reports:
- https://github.com/abrensch/brouter/issues/734

<img src="https://github.com/user-attachments/assets/cdfb128f-9bdb-49c7-823a-5d825be0f52c" width="50%"/>

**Reply by Ciutnik** [reply]
This is good news for motorway users 👍 

Perhaps it should be similar for exits from other road categories? All of them have the possibility of mapping the appropriate "xxx_link". Although this probably applies mostly to "trunk" and "primary".

**Reply by devemux86** [reply]
@Ciutnik 

Please post any suggestions in the BRouter repository. The instructions come from BRouter. :slightly_smiling_face: 
- https://github.com/abrensch/brouter/issues/734

But first better wait to test this new BRouter feature.

**Reply by Ciutnik** [reply]
Agreed. That was just my loose thought. Surely the BRouter developers thought about it 🙂

**Reply by devemux86** [reply]
Note that few routers have "exit" instructions:
BRouter and only Valhalla, OSRM, Google Maps.

**Reply by devemux86** [reply]
BRouter "exit" instructions are available in **[Cruiser 5.1.0-b7](https://github.com/devemux86/cruiser/discussions/671)**.

**Comment by devemux86** [comment, marked as answer]
- BRouter (offline) exit left / right instructions and other turn instruction improvements are available in https://github.com/devemux86/cruiser/discussions/684
